tests/client/crud_spec/framework.rs

```diff
@@ -170,7 +170,7 @@ macro_rules! run_insert_one_test {
 macro_rules! run_replace_one_test {
     ( $db:expr, $coll:expr, $filter:expr, $replacement:expr, $upsert:expr,
         $outcome:expr ) => {{
-            let options = ReplaceOptions::new($upsert, None);
+            let options = ReplaceOptions { upsert: $upsert, write_concern: None };
```

Comment:
Yeah, I think I probably only implemented all this in macros at the time because I had finally gotten to the point where I understood them and therefore wanted to celebrate by using them everywhere.

Let's open an issue to convert the test frameworks from macros to legitimate Rust code; it'll be a bit of work, but it'll probably save a lot of hassle in the long run.